configs/settings-example.yml

```diff
@@ -0,0 +1,55 @@
+# Variables for ansible provisioning
+#  https://docs.ansible.com/ansible/latest/playbooks_variables.html
+
+env: "dev" # dev | production - to enable TLS change mediawiki.protocal to 'https' below
+#apt_proxy: IP address # Install apt-cacher-ng on the host to avoid downloading packages again.
+
+mediawiki:
+  domain: hitchwiki.test
+  protocol: http # change to 'https' to enable TLS
+  debug: False
+  cache: False
+  branch: REL1_29
+  sitename: Hitchwiki
+  metanamespace: Hitchwiki
+  cookiedomain: .hitchwiki.test
+  db_prefix: "hw_"
+  protocol: http # change to https with enabled TLS
+  owner: "{{ apache.user }}:{{ apache.group }}"
+  db_prefix: "hw_"
+  visible_to_search_engines: True
+  user:
+    Hitchwiki: "{{ db.phpmyadmin_password }}"
+    Hitchbot: "{{ db.phpmyadmin_password }}"
+    Hitchhiker: "{{ db.phpmyadmin_password }}"
+  smtp: # MailDev is running at port 1025 at dev server https://github.com/djfarrelly/MailDev
+    enabled: true # See http://hitchwiki.test:1080/
+    host: localhost
+    port: 1025
+    auth: false # Should we use SMTP authentication (true or false)
+    username:
+    password:
+  spam: # https://www.google.com/recaptcha/intro/index.html
+    spamprotection: False
+    # These keys will work only at domains hitchwiki.test and 192.168.33.10
+    # TODO use vault https://docs.ansible.com/ansible/latest/playbooks_vault.html
+    recaptchapublickey: 6Lfx_-cSAAAAAFNMDH9fuFc37x_TYkOAcYHY_ZNc
+    recaptchaprivatekey: 6Lfx_-cSAAAAAAdawoKHU7z83VS3aEJETODcCLQf
+  mapbox:
+    username: trustroots
+    # Public access token
+    access_token: pk.eyJ1IjoidHJ1c3Ryb290cyIsImEiOiJVWFFGa19BIn0.4e59q4-7e8yvgvcd1jzF4g
+    mapkey_streets: ce8bb774
+    mapkey_satellite: kil7hee6
+  geonames:
+    username: hitchwiki
+    # use these instead of the above if geonames requests are being throttled
+    # during migration. keep this list in the end of the file, as "[]" causes
+    # bash_ini_parser to crash (we don't need these variables in bash)
+    alternatives: { hitchwiki, hitchwiki2, hitchwiki3 }
+
+vagrant:
+  private_network_ip: "192.168.33.10" # Sets IP address for the Virtual Machine
+  hostname: "{{ mediawiki.domain }}" # Hostname to use for the development server
```

Comment:
Since these settings are read by Vagrant at this point, this won't work (`{{ mediawiki.domain }}` is an Ansible feature?) Normally yaml files can't have logic like this.

Now Vagrant throws:
```
There are errors in the configuration of this machine. Please fix
the following errors and try again:

vm:
* The hostname set for the VM should only contain letters, numbers,
hyphens or dots. It cannot start with a hyphen or dot.
```

Let's move the hostname to top level since it's a shared setting used by Vagrant, Mediawiki and Apache.

Reply:
I assume  `domain` is identical to `hostname`, sw we can get rid of mediawiki.domain.